### DISCUSSION TITLE
Is it possible to assign a function to keymaps in Lspsaga finder config ?

### DISCUSSION THREAD

**Discussion by eshepelyuk** [nvimdev/lspsaga.nvim#1495] [Q&A]
Hello

I would like to use this plugin https://github.com/nvim-focus/focus.nvim to open LSP references files 
insted of using native `split`.
Is it possible to assign a function to `finder.keys ` so I could call other plugin funcions.

Smth like
```lua
require('lspsaga').setup({
  finder = {
    keys = {
      on_open = function(params_with_file_path) 
      .....
      end
    }
  }
})
```

**Comment by jaidetree** [comment]
Proposed a possible solution in the finder tool https://github.com/nvimdev/lspsaga.nvim/issues/1523 if you'd like to weigh in.

**Comment by jaidetree** [comment]
After for poking around there is a way to achieve that behavior with the existing APIs.

First, you will need to define a custom command like the following:

```lua
vim.api.nvim_create_user_command(
  "MyCustomCommand",
  function (opts)
    local filepath = opts.fargs[1]
    print("MyCustomCommand filepath=", filepath)
  end,
  {
    nargs = 1,
    complete = "file",
  }
) 
```

Then in the setup you can use the following:

```lua
require('lspsaga').setup({
  finder = {
    keys = {
      "open_or_toggle" = "p",
      "MyCustomCommand" = "o",
    },
  },
})
```

That will fire your custom command function if you press `o` in the lspsaga finder window. That said I don't love having to define commands just for this one plugin, I've drafted #1525 so it's more direct without breaking the original `keys` api contract.
